src/externalized/mercs/MercProfile.cc

```diff
@@ -90,4 +91,409 @@ MERCPROFILESTRUCT &MercProfile::getStruct() const
 MercProfile::operator MERCPROFILESTRUCT&() const
 {
 	return *m_profile;
-}
\ No newline at end of file
+}
+
+void MercProfile::deserializeStructRelations(MERCPROFILESTRUCT* prof, const JsonObject& json, const ContentManager* contentManager)
+{
+	for (auto& element : json["relations"].toVec()) {
+		JsonObject r = element.toObject();
+		uint8_t idx = contentManager->getMercProfileInfoByName(r.GetString("target"))->profileID;
+		if (idx >= NUM_RECRUTABLE) {
+			throw DataError(ST::format("{}'s profile id must be less than {}", r.GetString("target"), NUM_RECRUTABLE));
+		}
+		prof->bMercOpinion[idx] = r.GetInt("opinion");
+		if (r.getOptionalBool("friend1")) {
+			prof->bBuddy[BUDDY_SLOT1] = idx;
+		}
+		else if (r.getOptionalBool("friend2")) {
+			prof->bBuddy[BUDDY_SLOT2] = idx;
+		}
+		else if (r.getOptionalBool("eventualFriend")) {
+			prof->bLearnToLike = idx;
+			prof->bLearnToLikeTime = r.getOptionalUInt("resistanceToBefriending");
+		}
+		else if (r.getOptionalBool("enemy1")) {
+			prof->bHated[HATED_SLOT1] = idx;
+			prof->bHatedTime[HATED_SLOT1] = r.getOptionalUInt("tolerance");
+		}
+		else if (r.getOptionalBool("enemy2")) {
+			prof->bHated[HATED_SLOT2] = idx;
+			prof->bHatedTime[HATED_SLOT2] = r.getOptionalUInt("tolerance");
+		}
+		else if (r.getOptionalBool("eventualEnemy")) {
+			prof->bLearnToHate = idx;
+			prof->bLearnToHateTime = r.getOptionalUInt("resistanceToMakingEnemy");
+		}
+	}
+}
+
+MERCPROFILESTRUCT* MercProfile::deserializeStruct(const JsonObject& json, const ContentManager* contentManager)
+{
+	MERCPROFILESTRUCT* prof = new MERCPROFILESTRUCT();
```

Comment:
Please use unique_ptr here and return that from this function. It's preferable to make ownership of objects more obvious.